core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -670,6 +680,13 @@ public CreatePathResult createPath(RpcContext rpcContext, LockedInodePath inodeP
         // create the new inode, with a write lock
         if (options instanceof CreateDirectoryOptions) {
           CreateDirectoryOptions directoryOptions = (CreateDirectoryOptions) options;
+          if (options.isInheritParentPermissions()) {
+            // inherit the parent directory permissions
+            directoryOptions.setOwner(currentInodeDirectory.getOwner())
```

Comment:
After loading metadata with inherit parent option, will a user gain access to a file that is not accessible by that user on UFS?